### ISSUE TITLE
Add ignore: ["comments"] option to max-empty-lines

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#2356]
Ref: https://github.com/stylelint/stylelint/issues/2343

**Comment by sendilkumarn** [comment]
I will give this one a shot

**Comment by jeddy3** [comment]
> I will give this one a shot

Awesome. 

We should quickly clarify the workings of this option before you start on it. Mainly, are we proposing this option would ignore _all_ comments or just "non"-shared-line comments (FYI, in stylelint we call comments that do not live on a line _by themselves_ "shared-line comments", but I don't think we have a name for comments that do live on a line by themselves). 

Given the following config, what warning do you think should be flagged?

```js
max-line-length: [15, { ignore: ["comments"] }]
```

```css
/* single line non-shared-line comment */

/* 
 * multi-line
 * non-shared-line comment
 */

a {
  color: red; /* shared-line comment that pushes line over 15 */
}
```

Depending on what we decide, we might need to be more specific with the option keyword.

**Comment by evilebottnawi** [comment]
@jeddy3 in theory we can remove all comments (all, only shared or how do we decide) and then check source code, it is not difficult task :smile:

**Comment by jeddy3** [comment]
Ignoring _all_ comments is fine with me. Just wanting to check that we all think this would be the _most useful_ option to provide users.

**Comment by sendilkumarn** [comment]
👍  *all comments*

For your case, we should have `no-warning`

**Comment by evilebottnawi** [comment]
@jeddy3 i think `all` is right way, btw postcss have pure support shared comments, we will can resolve this after postcss 6.0 released (or review if it is requested by people).

**Comment by sendilkumarn** [comment]
between is it `max-line-length` or `max-empty-lines` or `both`

**Comment by evilebottnawi** [comment]
@sendilkumarn need create separate issue for this request :smile:

**Comment by jeddy3** [comment]
> between is it max-line-length or max-empty-lines or both

Sorry, my bad. I totally missed that this issue was for `max-empty-lines`. I'll switch this issue to `max-line-length` and create another issue for `max-empty-lines.

**Comment by jeddy3** [comment]
> between is it max-line-length or max-empty-lines or both

Sorry, my bad. I'm talking about completely the wrong rule. [`max-line-length` already has an `ignore: ["comments"]` option](https://stylelint.io/user-guide/rules/max-line-length/#ignore-comments), which targets _all_ comments. Forget my comment above: https://github.com/stylelint/stylelint/issues/2356#issuecomment-282688602

This issue is about `max-empty-lines`, and adding an option to ignore _all_ comments. 

@sendilkumarn Take it away :)
